### PR TITLE
Validate `getConfigPath()` return value in `createConfigContext()`

### PR DESCRIPTION
**Pull request by dahlia**
## Summary

Fixes https://github.com/dahlia/optique/issues/416

`createConfigContext()` already validates that `getConfigPath` is a function, but it never checked the return value at runtime. When `getConfigPath()` returns a non-string value (an object, a Promise, a number, etc.), the value is passed directly to `path.resolve()`, which crashes with a raw `TypeError [ERR_INVALID_ARG_TYPE]` from Node internals instead of a clear Optique error.

This PR adds a runtime type check on the `getConfigPath()` return value, throwing a descriptive `TypeError` if it is not a `string` or `undefined`:

```typescript
const configPath = opts.getConfigPath(parsedPlaceholder);

if (configPath != null && typeof configPath !== "string") {
  throw new TypeError(
    `Expected getConfigPath() to return a string or undefined, but got: ${
      getTypeName(configPath)
    }.`,
  );
}
```

This follows the same validation pattern already used for the `getConfigPath`, `load`, `fileParser`, and `schema` options elsewhere in the config package.

## Test plan

- Added three regression tests in *packages/config/src/index.test.ts* covering object, Promise, and number return values from `getConfigPath()`
- All tests written before the fix (TDD) and confirmed to fail, then pass after the fix
- Full test suite passes (`mise test`)

### CODE REVIEW

**Comment by dahlia** [comment]
@codex review

File: packages/config/src/index.ts [hunk 1]
```diff
@@ -919,6 +919,14 @@ export function createConfigContext<T, TConfigMeta = ConfigMeta>(
         // Single-file mode
         const configPath = opts.getConfigPath(parsedPlaceholder);
 
+        if (configPath != null && typeof configPath !== "string") {
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/config/src/index.ts]: Fixed in 8c5559a — changed `configPath != null` to `configPath !== undefined` so that `null` is also rejected, and added a dedicated test case for `null` returns.

File: packages/config/src/index.ts [hunk 2]
```diff
@@ -919,6 +919,14 @@ export function createConfigContext<T, TConfigMeta = ConfigMeta>(
...
+          throw new TypeError(
+            `Expected getConfigPath() to return a string or undefined, but got: ${
+              getTypeName(configPath)
+            }.`,
+          );
+        }
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/config/src/index.ts]: Fixed in 8c5559a — changed `configPath != null` to `configPath !== undefined` so that `null` is also rejected, and added a dedicated test case for `null` returns.

**Comment by dahlia** [comment]
@codex review
